plugin/builtin/helmchartinflationgenerator/HelmChartInflationGenerator_test.go

```diff
@@ -626,3 +627,180 @@ func copyTestChartsIntoHarness(t *testing.T, th *kusttest_test.HarnessEnhanced)
+metadata:
+ name: %s
+name: %s
+version: %s
+repo: %s
+releaseName: %s
+`, tt.chartName, tt.chartName, tt.version, tt.repo, tt.releaseName)
+
+			rm := th.LoadAndRunGenerator(config)
+			assert.True(t, len(rm.Resources()) > 0)
+
+			var chartDir string
+			if tt.version != "" && tt.repo != "" {
+				chartDir = fmt.Sprintf("charts/%s-%s/%s", tt.chartName, tt.version, tt.chartName)
+			} else {
+				chartDir = fmt.Sprintf("charts/%s", tt.chartName)
+			}
+
+			fmt.Printf("%s: %s\n", tt.name, chartDir)
+
+			d, err := th.GetFSys().ReadFile(filepath.Join(th.GetRoot(), chartDir, "Chart.yaml"))
+			if err != nil {
+				t.Fatal(err)
+			}
+
+			assert.Contains(t, string(d), fmt.Sprintf("name: %s", tt.chartName))
+			if tt.version != "" {
+				assert.Contains(t, string(d), fmt.Sprintf("version: %s", tt.version))
+			}
+		})
+	}
+}
+
+// Test that verifies +1 instances of same chart with different versions
+// https://github.com/kubernetes-sigs/kustomize/issues/4813
+func TestHelmChartInflationGeneratorWithMultipleInstancesSameChartDifferentVersions(t *testing.T) {
+	th := kusttest_test.MakeEnhancedHarnessWithTmpRoot(t).
+		PrepBuiltin("HelmChartInflationGenerator")
+	defer th.Reset()
+	if err := th.ErrIfNoHelm(); err != nil {
+		t.Skip("skipping: " + err.Error())
+	}
+
+	podinfo1 := th.LoadAndRunGenerator(`
+apiVersion: builtin
+kind: HelmChartInflationGenerator
+metadata:
+ name: podinfo
+name: podinfo
+version: 6.2.1
+repo: https://stefanprodan.github.io/podinfo
+releaseName: podinfo1
+`)
+
+	podinfo2 := th.LoadAndRunGenerator(`
+apiVersion: builtin
+kind: HelmChartInflationGenerator
+metadata:
+ name: podinfo
+name: podinfo
+version: 6.1.8
+repo: https://stefanprodan.github.io/podinfo
+releaseName: podinfo2
+`)
+
+	podinfo1Img, err := podinfo1.Resources()[1].GetFieldValue("spec.template.spec.containers.0.image")
+	assert.NoError(t, err)
+	assert.Equal(t, "ghcr.io/stefanprodan/podinfo:6.2.1", podinfo1Img)
+
+	podinfo2Img, err := podinfo2.Resources()[1].GetFieldValue("spec.template.spec.containers.0.image")
+	assert.NoError(t, err)
+	assert.Equal(t, "ghcr.io/stefanprodan/podinfo:6.1.8", podinfo2Img)
+
+	podinfo1ChartsDir := filepath.Join(th.GetRoot(), "charts/podinfo-6.2.1/podinfo")
+	assert.True(t, th.GetFSys().Exists(podinfo1ChartsDir))
+
+	podinfo2ChartsDir := filepath.Join(th.GetRoot(), "charts/podinfo-6.1.8/podinfo")
+	assert.True(t, th.GetFSys().Exists(podinfo2ChartsDir))
+
+	podinfo1ChartContents, err := th.GetFSys().ReadFile(filepath.Join(podinfo1ChartsDir, "Chart.yaml"))
+	assert.NoError(t, err)
+	assert.Contains(t, string(podinfo1ChartContents), "version: 6.2.1")
+
+	podinfo2ChartContents, err := th.GetFSys().ReadFile(filepath.Join(podinfo2ChartsDir, "Chart.yaml"))
+	assert.NoError(t, err)
+	assert.Contains(t, string(podinfo2ChartContents), "version: 6.1.8")
+}
+
+// Addressed: https://github.com/kubernetes-sigs/kustomize/issues/5163
+func TestHelmChartInflationGeneratorWithLocalChartWithVersion5163(t *testing.T) {
```

Comment:
IMO, we shouldn't specify which issue we solve in the function name.

Reply:
actually, after rebasing it from the master branch, I took the same approach as the existing unit test, here https://github.com/kubernetes-sigs/kustomize/blob/aeb7cb2a4040c19fa0771240b65832c0ba046a21/plugin/builtin/helmchartinflationgenerator/HelmChartInflationGenerator_test.go#L585
wdyt?

Reply:
let's wait for other reviewer's opinion

Reply:
I think a better test name is meaningful and easy to understand what tests.
Maybe you can write what we test for the test name without the issue number, but I want to keep the issue links in the comment above function.

Reply:
sure, let me figure a better name